RecoTauTag/RecoTau/src/PFRecoTauClusterVariables.cc

```diff
@@ -83,8 +83,16 @@ namespace reco {
         if (signal_pfcand != nullptr) {
           ecal_en_in_signal_pf_cands += signal_pfcand->ecalEnergy();
           hcal_en_in_signal_pf_cands += signal_pfcand->hcalEnergy();
+        } else {
+          // TauReco@MiniAOD: individual ECAL and HCAL energies recovered from fractions
+          const pat::PackedCandidate* signal_pcand = dynamic_cast<const pat::PackedCandidate*>(signal_cand.get());
+          if (signal_pcand != nullptr) {
```

Comment:
```suggestion
```
remove a check, or add an assert: if we are here and it's not a packedCand, it's most likely that we have a bug